src/clj/fluree/db/query/exec/where.cljc

```diff
@@ -566,6 +566,23 @@
                           clause-ch)
     out-ch))
 
+(defmethod match-pattern :values
+  [db fuel-tracker solution pattern error-ch]
+  (let [[_ inline-solutions] pattern
+        ;; need to compute sids on inline data so they can match equal solutions
+        inline-solutions* (mapv (fn [solution]
+                                  (->> solution
+                                       (mapv (fn [[var match]]
+                                               [var (if (matched-iri? match) (compute-sid match db) match)]))
+                                       (into {} )))
+                                inline-solutions)]
+    ;; filter out any solutions that don't match inline solution data
+    (if (every? (fn [inline-solution] (= (select-keys solution (keys inline-solution))
```

Comment:
I don't think this will work in all cases. There could be other metadata that we pull from the dbs that won't be in the inline solution. There might also be problems in the context of federated queries where a subject was pulled from a different database than the one being evaluated now, so the sid that you compute above won't match. I think you have to pull the value or iri out specifically instead of relying on the entire match map to be equal with the inline one for each inline variable.

Reply:
Hmm, maybe `:lang` from :meta as well? I'm not sure about `:i`, though.